ownCloud/Viewer/PDFOutlineViewController.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  PDFOutlineViewController.swift
+//  ownCloud
+//
+//  Created by Michael Neuwert on 04.10.2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2018, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+import UIKit
+import PDFKit
+
+class PDFOutlineViewController: UIViewController, Themeable {
+
+    enum Mode : Int {
+        case ToC, Thumbnails
+    }
+
+    var pdfDocument: PDFDocument?
+    var themeCollection: ThemeCollection?
+
+    var mode: Mode = .ToC {
+        didSet {
+            if let control = modeSegmentedControl {
+                control.selectedSegmentIndex = mode.rawValue
+            }
+            setupChildController(forMode: mode)
+        }
+    }
+
+    var modeSegmentedControl: UISegmentedControl?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        var iconArray: [UIImage] = [UIImage]()
+        if self.pdfDocument?.outlineRoot != nil {
+            iconArray.append( UIImage(named: "ic_pdf_outline")!)
+        }
+        iconArray.append(UIImage(named: "ic_pdf_view_multipage")!)
+
+        modeSegmentedControl = UISegmentedControl(items: iconArray)
+        if self.pdfDocument?.outlineRoot != nil {
+            modeSegmentedControl?.addTarget(self, action: #selector(modeChanged), for: .valueChanged)
+            self.mode = .ToC
+        } else {
+            self.mode = .Thumbnails
+        }
+        self.navigationItem.titleView = modeSegmentedControl
+
+        if UIDevice.current.userInterfaceIdiom != .pad {
+            let resumeItem = UIBarButtonItem(title: "Resume".localized, style: .plain, target: self, action: #selector(resume))
+            self.navigationItem.rightBarButtonItem = resumeItem
+        }
+
+        Theme.shared.register(client: self, applyImmediately: true)
+
+    }
+
+    deinit {
+        Theme.shared.unregister(client: self)
+        NotificationCenter.default.removeObserver(self)
```

Comment:
This might be left-over code: I can't find any prior `addObserver` calls.

Reply:
@felix-schwarz fixed